### ISSUE TITLE
[Main & MIEB] potential issues for multi-label classification

### ISSUE THREAD

**Issue by gowitheflow-1998** [embeddings-benchmark/mteb#1835]
@isaac-chung and I were debugging the weird scores for `VOC2007` in `mieb` (#1792) and found the following potential issues for both `mieb` and `main`.

1. `lrap` computation. Think `lrap` is supposed to operate on continuous scores instead of discrete predicted labels which I fixed for MIEB here #1834. This change largely smooths the large performance range across models when the `samples_per_label` is small.
Was trying to fix the same for `main` but got a chain of failed tests. Note that a difference is `main` is using `KNeighborsClassifier()` without a `MultiOutputClassifier` which I am not sure works in the expected way either; `mieb` instead is using `MultiOutputClassifier(estimator=LogisticRegression())` which treats each label as a separate binary cls problem. Will be great if someone can confirm the same for main! Also, why KNeighbors?

2. Logic for under sampling for `samples_per_label`. Both `main` and `mieb` are undersampling with a per single-label logic instead of a per set-label logic which I am not sure is what we want either. i.e.,
```python
if any((label_counter[label] < samples_per_label) for label in y[i]):
    sample_indices.append(i)
```
With a `samples_per_label=8` we can get something like the following, where some labels have much more number of examples than the other. e.g., for an example of label (18, 14), if the number of 18th < 8, even though the 14th is already 40+, this example will still be sampled and add to the training set.
```
defaultdict(int,
            {14: 49,
             13: 8,
             4: 12,
             10: 8,
             5: 9,
             2: 8,
             11: 8,
             6: 11,
             7: 9,
             18: 8,
             12: 8,
             19: 9,
             17: 9,
             8: 8,
             0: 8,
             3: 8,
             15: 8,
             9: 8,
             16: 8,
             1: 8})
```
Think this is less of a problem if using `MultiOutputClassifier()` because it only means that certain binary classifiers are trained with more examples while others are not affected.

3. The number of unique predicted labels are much fewer than unique groundtruth labels, which can make `accuracy` much lower because it's currently assessing in a strict set-label perfect matching logic. e.g., `[0, 0, 0, 1, 0]` will score `0` for `[0, 0, 0, 1, 1]` which is probably not optimal.

![Image](https://github.com/user-attachments/assets/530918ce-82bf-4c33-8fac-5556e700a815)

![Image](https://github.com/user-attachments/assets/409aa5fd-a198-44fd-9e5a-b7aff6ed4d58)

**Comment by gowitheflow-1998** [comment]
@isaac-chung @KennethEnevoldsen @Samoed

**Comment by isaac-chung** [comment]
Was discussing with @Samoed offline. We should bump the AbsTask to MultilabelClassification.v2 to fix this, and at the same time preserve reproducibility for all previous tasks. 

We should double check, as I think we had fixed this in MIEB, which means they should have been using this hypothetical v2.

**Comment by Samoed** [comment]
Mieb and mteb have the same code https://github.com/embeddings-benchmark/mteb/blob/721e8a34743c050179ee5e9093ed24ee7ab37fba/mteb/abstasks/AbsTaskMultilabelClassification.py#L248-L260
https://github.com/embeddings-benchmark/mteb/blob/721e8a34743c050179ee5e9093ed24ee7ab37fba/mteb/abstasks/Image/AbsTaskImageMultilabelClassification.py#L285-L297

**Comment by isaac-chung** [comment]
Hi all @KennethEnevoldsen @Samoed @gowitheflow-1998 : 

For 2): So the code was left as is. However, the same problem persists. I think we should go with this fix as previously mentioned:
> We should bump the AbsTask to MultilabelClassification.v2 to fix this, and at the same time preserve reproducibility for all previous tasks.

For 3) : 
> The number of unique predicted labels are much fewer than unique groundtruth labels

I added pdb on L170 of `mteb/abstasks/multilabel_classification.py`, ran `MalteseNewsClassification`, and saw this:

<img width="300" height="91" alt="Image" src="https://github.com/user-attachments/assets/165747fd-cba7-4aab-ba78-45b66dae6308" />

**Comment by KennethEnevoldsen** [comment]
I think this sounds very reasonable. Will you make the fix @isaac-chung? 

(re bump MultilabelClassification.v2 it might be nice to allow it to support non-multilabel as well in an effort to merge it with CLF)

**Comment by isaac-chung** [comment]
I'll think of a fix for 3) at least

**Comment by isaac-chung** [comment]
For 3, we _could_ implement hamming score as suggested in [this stackoverflow post](https://stackoverflow.com/questions/32239577/getting-the-accuracy-for-multi-label-prediction-in-scikit-learn).

**Comment by KennethEnevoldsen** [comment]
That sounds very reasonable @isaac-chung

**Comment by isaac-chung** [comment]
For 2), maybe we can add a comment to say that we will ensure at least one of the labels to have sample_per_label? This and adding the hamming score would not require bumping the abstask to v2.

**Comment by KennethEnevoldsen** [comment]
Yeah that sounds very decent

**Comment by isaac-chung** [comment]
Sweet. I'll start a PR based on this 💃
